compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -682,6 +738,77 @@ impl Type {
         }
     }
 
+    /// TODO(https://github.com/noir-lang/noir/issues/5156): Remove with explicit numeric generics
+    pub fn find_numeric_type_vars(&self, found_names: &mut Vec<String>) {
+        // Return whether the named generic has a TypeKind::Numeric and save its name
+        let named_generic_is_numeric = |typ: &Type, found_names: &mut Vec<String>| {
+            if let Type::NamedGeneric(_, name, TypeKind::Numeric) = typ {
+                found_names.push(name.to_string());
+                true
+            } else {
+                false
+            }
+        };
+
+        match self {
+            Type::FieldElement
+            | Type::Integer(_, _)
+            | Type::Bool
+            | Type::Unit
+            | Type::Error
+            | Type::TypeVariable(_, _)
+            | Type::Constant(_)
+            | Type::NamedGeneric(_, _, _)
```

Comment:
Looks like the TypeVariable and NamedGeneric cases were missed here. Without the later I'm guessing this never actually pushes any names.

Reply:
This looks to act the same as `contains_numeric_typevar`. We do expect it to push names wherever `named_generic_is_numeric` is called such as for the length in `Type::Array(length, elem)`. For a TypeVariable case I'm not sure what we would be attempting to push. In the NamedGeneric case I could call `named_generic_is_numeric` on `self`, but I'm mainly concerned about the array length here.

Reply:
I think you'll still need to check for any type variable bindings (or named generic bindings)

Reply:
Shouldn't we only be binding named generics during monomorphization?

Reply:
Yes, but type variables may be bound any time, and we may as well check named generics. It's just safer to include the check in case there's a case we miss or the method is later used in a different scenario.

Reply:
Ok I have added a check for type variable bindings and name generics